### PR TITLE
Debian 11 support and unattended installation

### PR DESCRIPTION
**Pull request by ajanvrin**
This PR implements the following features in the installation script (setup/install.sh):
 - Debian 11 support
 - Unattended mode (a "-y" flag that, when specified, automatically select optional dependencies for installation)

### CODE REVIEW

**Comment by HugoMskn** [comment]
Thanks, I was having issue with my debian server !

**Review by HugoMskn — APPROVED** [review]
Work perfectly fine with a Debian 11 server

**Comment by Cx01N** [comment]
Unattended mode is something I never realized I needed, thank you!

File: setup/install.sh [hunk 1]
```diff
@@ -111,27 +148,31 @@ else
 fi
 
 echo -e "\x1b[1;34m[*] Installing dotnet for C# agents and modules\x1b[0m"
-if [ $OS_NAME == "DEBIAN" ]; then
+if [ "$OS_NAME" == "DEBIAN 1*" ]; then
   wget https://packages.microsoft.com/config/debian/10/packages-microsoft-prod.deb -O packages-microsoft-prod.deb
```
Comment (vinnybod) [inline setup/install.sh:152]: does this need to differentiate between `debian/10` and `debian/11`?

Comment (vinnybod) [inline setup/install.sh:152]:
```suggestion
if [ "$OS_NAME" == "DEBIAN 10" ]; then
  wget https://packages.microsoft.com/config/debian/10/packages-microsoft-prod.deb
elif [ "$OS_NAME" == "DEBIAN 11" ]; then
  wget https://packages.microsoft.com/config/debian/11/packages-microsoft-prod.deb
fi
if [ "$OS_NAME" == "DEBIAN 1*" ]; then
```

Comment (ajanvrin) [reply setup/install.sh:152]: It does not ?
"DEBIAN 1*" matches both "DEBIAN 10" and "DEBIAN 11", as well as future versions

Comment (vinnybod) [reply setup/install.sh:152]: There should be `https://packages.microsoft.com/config/debian/11/packages-microsoft-prod.deb` in addition to `https://packages.microsoft.com/config/debian/10/packages-microsoft-prod.deb`

**Comment by vinnybod** [comment]
I included your commits in https://github.com/BC-SECURITY/Empire/pull/639 which finished up the code review comments, adds tests, and ubuntu 22.04.

**Pull request closed without merging**